app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/SurveyLocalDataSource.java

```diff
@@ -44,4 +45,13 @@ public void getUnsentSurveys(IDataSourceCallback<List<Survey>> callback) {
         }
         callback.onSuccess(unsentSurveys);
     }
+
+    @Override
+    public void updateAllQuarantineSurveysAsComplete() {
```

Comment:
the responsibilities of a repository is getting and save in database and map from db model to domain entity but have no responsibility for change survey status. I think that this responsibility is a domain responsibility then the repository should has a GetAllQuantineSurveys method and return domain survey entities then the PushUseCase iterate surveys and change status and invoke save method in the repository.